d3/d3.d.ts

```diff
@@ -119,7 +119,7 @@ declare module D3 {
         * @param arr Array to search
         * @param map Accsessor function
         */
-        extent<T, U>(arr: T[], map: (v: T) => U): U[];
+        extent<T, U>(arr: T[], map: (v?: T, i?: number) => U): U[];
```

Comment:
Ok I saw the previous link you posted.  This seems to be a bug on D3 since it doesn't fully wrap the map logic it seems.  Keeping it with 2 arguments seems fine.  However, you will still need to modify the other signatures that use the map idea.